src/bun.js/bindings/BunProcess.cpp

```diff
@@ -2315,17 +2322,27 @@ JSC_DEFINE_HOST_FUNCTION(Process_functionCpuUsage,
             }
 
             if (UNLIKELY(!userValue || !userValue.isNumber())) {
-                throwTypeError(globalObject, throwScope, "Expected a number for the user property"_s);
+                throwTypeError(globalObject, throwScope, "Expected a number for the 'user' property"_s);
                 return JSC::JSValue::encode(JSC::jsUndefined());
             }
 
             if (UNLIKELY(!systemValue || !systemValue.isNumber())) {
-                throwTypeError(globalObject, throwScope, "Expected a number for the system property"_s);
+                throwTypeError(globalObject, throwScope, "Expected a number for the 'system' property"_s);
                 return JSC::JSValue::encode(JSC::jsUndefined());
             }
 
-            double userComparator = userValue.asNumber();
-            double systemComparator = systemValue.asNumber();
+            double userComparator = userValue.toNumber(globalObject);
+            double systemComparator = systemValue.toNumber(globalObject);
+
+            if (userComparator > JSC::maxSafeInteger() || userComparator < 0 || std::isnan(userComparator)) {
```

Comment:
do we also need to check for `Infinity`?

Reply:
infinity is greater than JSC::maxSafeInteger